x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/Mode.java

```diff
@@ -33,4 +33,9 @@ public String toString() {
     public static boolean isDriver(Mode mode) {
         return mode == JDBC || mode == ODBC;
     }
+
+    // TODO: replace all "== Mode.CLI"?
+    public static boolean isCli(Mode mode) {
```

Comment:
In many cases the code look like `Mode.isDriver(mode) || mode == Mode.CLI` and `Mode.isDriver(mode) || Mode.isCli(mode)` looked more consistent to me. I got to it from a first thought of adding a `Mode.isStrictClient()` to check for all drivers+cli (since we treat those grouped in many cases). Irrespective of that, though, if Mode being an enum is to be considered, one would either stick to `mode == Mode.JDBC || mode == Mode.jdbc || mode == Mode.CLI` _or_ the two methods wrapping, IMO.
However, this isn't a strictly necessary refactoring, it is subjective and I'll happily change it back if you still think the old way is better (just 👍 this comment).